packages/core/entity/Tutorial/index.ts

```diff
@@ -0,0 +1,51 @@
+import { castArray } from "lodash";
+
+interface TutorialStep {
+    targetId: string;
+    message: string | React.ReactNode;
+}
+
+/**
+ * Entity responsible for containing information about a tutorial in which to guide
+ * a user through
+ */
+export default class Tutorial {
```

Comment:
Maybe add a comment somewhat similar to your explanation in the PR description that explains the targetID tagging thing.